examples/11_hens/utilities.py

```diff
@@ -0,0 +1,1050 @@
+    - It also ensures that `lat_min` is less than `lat_max`.
+    - For `lon_min` and `lon_max`, the function checks that `lon_min` is less than `lon_max`.
+    - Additionally, it verifies that `lon_min` and `lon_max` are within the valid range of -180 to 180 or 0 to 360, depending on whether `lon_min` is negative or positive.
+    """
+    if not (cfg_cropbox.lat_min >= -90):
+        raise ValueError("lat_min needs to be >=-90")
+    if not (cfg_cropbox.lat_max <= 90):
+        raise ValueError("lat_max needs to be <=90")
+    if not (cfg_cropbox.lat_min < cfg_cropbox.lat_max):
+        raise ValueError("lat_min needs to be smaller than lat_max")
+    if not (cfg_cropbox.lon_min < cfg_cropbox.lon_max):
+        raise ValueError("lon_max needs to be larger than lon_min")
+    if cfg_cropbox.lon_min < 0:
+        if not (cfg_cropbox.lon_min >= -180):
+            raise ValueError("lon_min needs to be >= -180")
+        if not (cfg_cropbox.lon_min < 180):
+            raise ValueError("lon_min needs to be < 180")
+        if not (cfg_cropbox.lon_max <= 180):
+            raise ValueError("lon_max needs to be <= 180")
+    else:
+        if not (cfg_cropbox.lon_min >= 0):
+            raise ValueError("lon_min needs to be >= 0")
+        if not (cfg_cropbox.lon_min < 360):
+            raise ValueError("lon_min needs to be < 360")
+        if not (cfg_cropbox.lon_max <= 360):
+            raise ValueError("lon_max needs to be <= 360")
+
+
+def determine_lon_ranges(
+    cfg_cropbox: DictConfig, lon_coords: np.ndarray[np.float64]
+) -> tuple[int, int]:
+    """
+    Determine the longitude ranges of the model and output based on the provided cropbox configuration and longitude coordinates.
+
+    Parameters
+    ----------
+    cfg_cropbox : DictConfig
+        Configuration object containing the cropbox settings, including `lon_min` and `lon_max`.
+    lon_coords : np.ndarray[np.float64]
+        A 1D array containing the longitude values of the grid, ordered ascending.
+
+    Returns
+    -------
+    lon_range_model : int
+        The range of the input longitude coordinates (either 180 or 360).
+    lon_range_out : int
+        The range of the desired output longitude coordinates (either 180 or 360).
+
+    Notes
+    -----
+    - The function determines whether the longitude coordinates of the model and the desired output range span from 0 to 180
+      or from -180 to 180 and converts accordingly.
+    - This is important for ensuring consistency in longitude ranges particularly when transitioning between different coordinate
+      systems or models.
+    """
+    # determine longitude ranges of input and output
+    lon_range_model = (
+        360 if ((lon_coords.max() > 180) and (lon_coords.min() >= 0)) else 180
+    )
+    lon_range_out = (
+        360 if ((cfg_cropbox.lon_max > 180) and (cfg_cropbox.lon_min >= 0)) else 180
+    )
+    return lon_range_model, lon_range_out
+
+
+def crop_area(
```

Comment:
the map_coords function should basically have this functionality If im following this correctly

https://github.com/NVIDIA/earth2studio/blob/main/earth2studio/utils/coords.py#L250

Reply:
Ignore, the -180:180 requirement makes it not usable